### DISCUSSION TITLE
Recursor returns unexpected SERVFAIL while resolving

### DISCUSSION THREAD

**Discussion by mayson** [PowerDNS/pdns#13775] [Q&A]
Hello,

We observe a strange situation: when trying to resolve a muiv.ru domain, SERVFAIL is returned. [I am attaching the trace log.](https://github.com/PowerDNS/pdns/files/14226298/trace.txt)

I believe that despite the fact that the DNS server ns2.miemp.ru was not resolved and polled, s_nsSpeeds.lock()->fastest returns 0ms, and thus this server is always has a priority. However domain miemp.ru has a DNS server ns2.muiv.ru... This cause a loop while trying to resolve of any DNS server record which stops by max-recursion-depth.

Are there any settings to avoid this situation?

The problem appeared after upgrade to 4.9.2 version, it seems that 4.7.6 doesn't have such a problem.

Thanks in advance.

**Comment by omoerbeek** [comment]
According to https://dnsviz.net/d/muiv.ru/dnssec/ this zone is  broken. A first step is to fix it and then see if Recursor is still having issues.
